src/pages/[platform]/build-a-backend/data/custom-business-logic/connect-eventbridge-datasource/index.mdx

```diff
@@ -0,0 +1,403 @@
+    },
+  },
+});
+```
+
+<Callout info>
+**NOTE:** To comply with the GraphQL spec, at least one query is required for a schema to be valid. Otherwise, deployments will fail a schema error. The Amplify Data schema is auto-generated with a `Todo` model and corresponding queries under the hood. You can leave the `Todo` model in the schema until you add the first custom query to the schema in the next steps.
+</Callout>
+
+## Step 2 - Add your Amazon EventBridge event bus as a data source
+
+In your `amplify/backend.ts` file, use the following code to add the default event bus as a data source for your API:
+
+```ts title="amplify/backend.ts"
+import { defineBackend } from "@aws-amplify/backend";
+import { auth } from "./auth/resource";
+import { data } from "./data/resource";
+import { aws_events } from "aws-cdk-lib";
+import {
+  Effect,
+  PolicyDocument,
+  PolicyStatement,
+  Role,
+  ServicePrincipal,
+} from "aws-cdk-lib/aws-iam";
+
+export const backend = defineBackend({
+  auth,
+  data,
+});
+
+// Create a new stack for the EventBridge data source
+const eventStack = backend.createStack("MyExternalDataSources");
+
+// Reference or create an EventBridge EventBus
+const eventBus = aws_events.EventBus.fromEventBusName(
+  eventStack,
+  "MyEventBus",
+  "default"
+);
+
+// Add the EventBridge data source
+// highlight-start
+backend.data.addEventBridgeDataSource("MyEventBridgeDataSource", eventBus);
+// highlight-end
+
+// Create a policy statement to allow invoking the AppSync GraphQL API
+const policyStatement = new PolicyStatement({
+  effect: Effect.ALLOW,
+  actions: ["appsync:GraphQL"],
+  resources: [`${backend.data.resources.graphqlApi.arn}/types/Mutation/*`],
+});
+
+// Create a role for the EventBus to assume
+const eventBusRole = new Role(eventStack, "AppSyncInvokeRole", {
+  assumedBy: new ServicePrincipal("events.amazonaws.com"),
+  inlinePolicies: {
+    PolicyStatement: new PolicyDocument({
+      statements: [policyStatement],
+    }),
+  },
+});
+
+// Create an EventBridge rule to route events to the AppSync API
+const rule = new aws_events.CfnRule(eventStack, "MyOrderRule", {
+  eventBusName: eventBus.eventBusName,
+  name: "broadcastOrderStatusChange",
+  eventPattern: {
+    source: ["amplify.orders"],
+    ["detail-type"]: ["OrderStatusChange"],
+    detail: {
+      orderId: [{ exists: true }],
+      status: ["PENDING", "SHIPPED", "DELIVERED"],
+      message: [{ exists: true }],
+    },
+  },
+  targets: [
+    {
+      id: "orderStatusChangeReceiver",
+      arn: backend.data.resources.cfnResources.cfnGraphqlApi
+        .attrGraphQlEndpointArn,
+      roleArn: eventBusRole.roleArn,
+      appSyncParameters: {
+        graphQlOperation: `
+        mutation PublishOrderFromEventBridge(
+          $orderId: String!
+          $status: String!
+          $message: String!
+        ) {
+          publishOrderFromEventBridge(orderId: $orderId, status: $status, message: $message) {
+            orderId
+            status
```

Comment:
Add "Attention: ..."

Reply:
added a callout because I didn't want to put a comment in the graphql string